hedera-node/hedera-token-service-impl/src/main/java/com/hedera/node/app/service/token/impl/handlers/TokenUpdateHandler.java

```diff
@@ -116,6 +119,27 @@ public void preHandle(@NonNull final PreHandleContext context) throws PreCheckEx
         if (op.hasAdminKey()) {
             context.requireKey(op.adminKeyOrThrow());
         }
+        // To update metadata either admin key or metadata key should sign.
```

Comment:
This is unrelated to this PR, but shouldn't we require the signatures of all new keys if they are changed?

Reply:
No any key changes **except** adminKey require only adminKey signature. But we have checks that keys are not empty and already exist on token.